ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/host/MicrofrontendPlatformRcpHost.java

```diff
@@ -77,20 +76,15 @@ public CompletableFuture<Browser> start(final MicrofrontendPlatformConfig config
         new Resource(Resources.get("js/helpers.js"), "application/javascript", "utf-8"))).start();
 
     // Create the browser and
-    hostBrowser = new Browser(shell, SWT.EDGE);
-    hostBrowser.addProgressListener(new ProgressAdapter() {
+    hostBrowser = BrowserViewFactory.createBrowserView(BrowserViewType.JXBROWSER, shell);
 
-      @Override
-      public void completed(final ProgressEvent event) {
-        startHost(config);
-      };
-    });
+    hostBrowser.addNavigationListener(() -> startHost(config));
 
     if (!headless) {
       shell.open();
     }
 
-    hostBrowser.setUrl(String.format("http://localhost:%d/host.html", Integer.valueOf(webserver.getPort())));
+    hostBrowser.loadUrl(String.format("http://localhost:%d/host.html", Integer.valueOf(webserver.getPort())));

```

Comment:
Please extract some variables. It is now a complete `((( )) (() ) ))) ` nightmare.